src/response_decoder.ts

```diff
@@ -150,30 +153,95 @@ function decodeDeliverResponse(dataResponse: DataReader, logger: Logger): Messag
 
 const EmptyBuffer = Buffer.from("")
 
-function decodeMessage(dataResponse: DataReader): Buffer {
+function decodeMessage(dataResponse: DataReader): Message {
   const messageLength = dataResponse.readUInt32()
   const startFrom = dataResponse.position()
 
   let content = EmptyBuffer
+  let type
+  let mapType
+  // let next = 0n
+  let headerType
+  let length
+  let applicationPropertiesLength = 0
+  let messageProperties: MessageProperties = {}
+  let applicationProperties: MessageApplicationProperties = {}
   while (dataResponse.position() - startFrom !== messageLength) {
     const formatCode = readFormatCodeType(dataResponse)
     switch (formatCode) {
       case FormatCodeType.ApplicationData:
         content = decodeApplicationData(dataResponse)
         break
-      case FormatCodeType.MessageAnnotations:
       case FormatCodeType.MessageProperties:
+        dataResponse.rewind(3)
+        type = dataResponse.readInt8()
+
+        if (type !== 0) {
+          throw new Error(`invalid composite header %#02x: ${type}`)
+        }
+
+        const nextType = dataResponse.readInt8()
+        switch (nextType) {
+          case FormatCode.SmallUlong:
+            dataResponse.readInt8() // read a SmallUlong
+            // next = BigInt(retSmallLong)
+            break
+          case FormatCode.ULong:
+            dataResponse.readUInt64() // read an ULong
+            // next = retULong
+            break
+          default:
+            // next = 0n
+            break
+        }
+        headerType = dataResponse.readUInt8()
+
+        switch (headerType) {
+          case FormatCode.List0:
+            length = 0
+            break
+          case FormatCode.List8:
+            dataResponse.forward(1)
+            const lenB = dataResponse.readInt8()
+            length = lenB
+            break
+          case FormatCode.List32:
+            dataResponse.forward(4)
+            const lenI = dataResponse.readInt32()
+            length = lenI
+            messageProperties = Properties.Parse(dataResponse, length)
+            break
+          default:
+            throw new Error(`ReadCompositeHeader Invalid type ${headerType}`)
+        }
+        break
```

Comment:
I suggest extracting this code in a function called `decodeMessageProperties` like the other branch of the switch